### PR TITLE
Update roadmap.md

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <vieux@docker.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @amylindburg @aluzzardi

File: ROADMAP.md [hunk 1]
```diff
@@ -1,29 +1,38 @@
 Docker Swarm Roadmap
 =====================
 
-####Security
-* [x] TLS authentication
-
 ####Scheduler
-* [ ] Persistent state storage
+Swarm comes with a builtin scheduler. It currently provides basic functionnalities, such as
```
Comment (thaJeztah) [inline ROADMAP.md]: builtin -> built-in?

s/functionnalities/functionalities

File: ROADMAP.md [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+scheduling containers based on constraints or affinity (co-scheduling of containers), persistant
```
Comment (thaJeztah) [inline ROADMAP.md]: s/persistant/persistent/

File: ROADMAP.md [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
+storage and multiple scheduling stragegies like binpacking or random.
```
Comment (thaJeztah) [inline ROADMAP.md]: s/stragegies/strategies/

File: ROADMAP.md [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+
+We plan to add more features to the builtin scheduler such as rebalancing (in case of host faillure)
```
Comment (thaJeztah) [inline ROADMAP.md]: built-in
s/faillure/failure/

File: ROADMAP.md [hunk 5]
```diff
@@ -1,29 +1,38 @@
...
+
 * [ ] Virtual Container ID
 * [ ] Rebalancing
-* [x] Affinity constraints & improved constraints expression (==, !=, regular expressions)
-* [ ] Global scheduling (schedule containers on every node)
+* [ ] Global scheduling
+
+####Leader Election (Distributed State)
+Regarding Swarm Multi-tenancy, we are working on shared states between multiple "soon-to-be-master"
+and  master election.
```
Comment (thaJeztah) [inline ROADMAP.md]: nit: two spaces between "and" and "master"

File: ROADMAP.md [hunk 6]
```diff
@@ -1,29 +1,38 @@
...
 
 ####Extensibility
-* [ ] Pluggable scheduler
-* [ ] Discovery backends
-  * [x]    etcd
-  * [x]    zookeeper
-  * [x]    consul
-  * [x]    hub 
-  * [x]    file
+The builtin scheduler allows you to manage container on apporximalty hundreds of machines within a cluster.
```
Comment (thaJeztah) [inline ROADMAP.md]: built-in
s/apporximalty/approximately/

File: ROADMAP.md [hunk 7]
```diff
@@ -1,29 +1,38 @@
...
+In the future, we want to allow third party tools to be plugged into Docker Swarm to allow management
+of much bigger clusters.
+
+The work has started on defining the "cluster" interface in [#393](https://github.com/docker/swarm/issues/393)
+The first integration will be Mesos but other tools will arrive after, like kubernetes.
```
Comment (thaJeztah) [inline ROADMAP.md]: perhaps links to Mesos and Kubernetes, for people unfamiliar with them?

**Comment by thaJeztah** [comment]
Marked some typos while reading, I realize it's not a real "docs", so probably no huge issue.

**Comment by vieux** [comment]
@thaJeztah thanks! Wrote that in a hurry, and didn't check well.

File: ROADMAP.md [hunk 8]
```diff
@@ -1,29 +1,38 @@
...
-  * [x]    zookeeper
-  * [x]    consul
-  * [x]    hub 
-  * [x]    file
+The builtin scheduler allows you to manage container on approximately hundreds of machines within a cluster.
+In the future, we want to allow third party tools to be plugged into Docker Swarm to allow management
+of much bigger clusters.
+
+The work has started on defining the "cluster" interface in [#393](https://github.com/docker/swarm/issues/393)
+qThe first integration will be Mesos but other tools will arrive after, like kubernetes.
```
Comment (thaJeztah) [inline ROADMAP.md]: stray `q`

**Comment by aluzzardi** [comment]
Wasn't the plan to stick that into the wiki?

**Comment by amylindburg** [comment]
How we have it modeled now is that we keep the ROADMAP.md doc as is, and use the Wiki to focus on the Releases — i.e., the subset of the Roadmap that will be targeted for each release.

**Comment by aluzzardi** [comment]
I'm worried about out of sync - I'm pretty sure it'll only take a couple of months before the two diverge. 

For instance, a PR to ROADMAP.md requires an update to the wiki after merging, which we'll eventually forget to do.

How about linking to the wiki instead?

**Comment by amylindburg** [comment]
Any change to the Roadmap -- regardless of where it lives -- will potentially result in a change to the current "Release Page" that we have imagined for the Wiki. This is the only page that should need updating if the Roadmap changes. If we are not keeping the "Release Page" in sync, it means that we are getting a bit sloppy with our Release Process.

The benefit of keeping ROADMAP.md is that you can submit a PR regarding changing the roadmap; this seems like a benefit.

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**